### PR TITLE
Cargo tug new sprite and fixes vehicles.

### PR DESCRIPTION
**Pull request by alsoandanswer**
What was supposed to be a 5 minute resprite has turned into a 5 day vehicle fix and overhaul.

- fixes vehicle latching
- clarifies vehicle mechanics
- fixes crate and closet sprites being loaded onto trolleys
- clarifies vehicle mechanics
- refactors janicart and some cleanup

### CODE REVIEW

**Comment by Greenjoe12345** [comment]
The north/south sprites of the new cargo tug seem a bit odd to be still facing left/right
It somehow didn't feel as bad with the old isometric style sprite of them

**Comment by atteria** [comment]
The tug's wagons _can't_ rotate because crates don't rotate.
And the 'somehow not as bad' feeling of the old cabinet projection sprites is just familiarity.

File: code/modules/vehicles/bike.dm [hunk 1]
```diff
@@ -120,7 +124,9 @@
 		return FALSE
 
 /obj/vehicle/bike/Move(var/turf/destination)
-	if(kickstand) return
+	if(kickstand)
+		visible_message("\The kickstand prevents the bike from moving!") // This nearly tricked me into thinking bikes were broken. Fuck me. -Wezzy
```
Comment (NonQueueingMatt) [inline code/modules/vehicles/bike.dm:128]:
```suggestion
		visible_message("The kickstand prevents the bike from moving!") // This nearly tricked me into thinking bikes were broken. Fuck me. -Wezzy
```

\The does not work if you don't have a var right next to it

File: code/modules/vehicles/cargo_train.dm [hunk 2]
```diff
@@ -243,48 +258,39 @@
...
-	turn_off()
-	if(!on)
-		to_chat(user, SPAN_NOTICE("You stop [src]'s engine."))
 
 /obj/vehicle/train/cargo/engine/proc/remove_key(var/mob/user)
 	if(!key)
 		to_chat(usr, SPAN_WARNING("\The [src] doesn't have a key inserted!"))
 		return
 	if(load && load != usr)
+		to_chat(usr, SPAN_WARNING("You can't remove \the [key] from \the [src] while its in use."))
```
Comment (NonQueueingMatt) [inline code/modules/vehicles/cargo_train.dm:290]:
```suggestion
		to_chat(usr, SPAN_WARNING("You can't remove \the [key] from \the [src] while it's in use."))
```

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]
Control clicking a cargo tug engine and then clicking "toggle latching" runtimes.

The cargo tug that spawns in Operations can't move at all. I suspect it has to do with something wonky in the way it latches things to itself when it spawns in, because spawning a new one works fine.

**Review by NonQueueingMatt — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

File: code/modules/vehicles/bike.dm [hunk 3]
```diff
@@ -120,7 +124,9 @@
 		return FALSE
 
 /obj/vehicle/bike/Move(var/turf/destination)
-	if(kickstand) return
+	if(kickstand)
+		visible_message("The kickstand prevents the bike from moving!") // This nearly tricked me into thinking bikes were broken. Fuck me. -Wezzy
```
Comment (SleepyGemmy) [inline code/modules/vehicles/bike.dm:128]:
```suggestion
		visible_message("The kickstand prevents the bike from moving!")
```

File: code/modules/vehicles/cargo_train.dm [hunk 4]
```diff
@@ -125,18 +130,19 @@
...
 
 /obj/vehicle/train/cargo/engine/attackby(obj/item/W as obj, mob/user as mob)
-	if(istype(W, /obj/item/key/cargo_train))
+	if(istype(W, key_type))
 		if(!key)
 			user.drop_from_inventory(W, src)
-			key = W
-			to_chat(user, SPAN_NOTICE("You slide the key into the ignition."))
+			key = W // put the key in the ignition
+			to_chat(user, SPAN_NOTICE("You slide \the [W] into \the [src]'s ignition."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/cargo_train.dm:142]:
```suggestion
			to_chat(user, SPAN_NOTICE("You slide \the [W] into \the [src]\s ignition."))
```

File: code/modules/vehicles/cargo_train.dm [hunk 5]
```diff
@@ -125,18 +130,19 @@
...
+			playsound(src, 'sound/machines/vehicles/key_in.ogg', 50, FALSE)
 		else
-			to_chat(user, SPAN_WARNING("\The [src] already has a key inserted."))
+			to_chat(user, SPAN_NOTICE("There is already a key in [src]'s ignition."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/cargo_train.dm:145]:
```suggestion
			to_chat(user, SPAN_NOTICE("There is already a key in \the [src]\s ignition."))
```

File: code/modules/vehicles/cargo_train.dm [hunk 6]
```diff
@@ -181,13 +187,23 @@
...
-/obj/vehicle/train/cargo/engine/turn_on()
+/obj/vehicle/train/cargo/engine/turn_on(var/mob/user)
 	if(!key)
 		audible_message("\The [src] whirrs, but the lack of a key causes it to shut down.")
 		return
-	else
+	if(!on)
 		..()
-		update_stats()
+		to_chat(user, SPAN_NOTICE("You turn on \the [src]'s ignition."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/cargo_train.dm:196]:
```suggestion
		to_chat(user, SPAN_NOTICE("You turn on \the [src]\s ignition."))
```

File: code/modules/vehicles/cargo_train.dm [hunk 7]
```diff
@@ -181,13 +187,23 @@
...
+		playsound(src, 'sound/machines/vehicles/button.ogg', 50, FALSE)
+		playsound_in(src, 'sound/machines/vehicles/start.ogg', 50, FALSE, time = 1 SECOND)
+	else
+		turn_off(user)
+	update_stats()
+
+/obj/vehicle/train/cargo/engine/turn_off(var/mob/user)
+	..()
+	to_chat(user, SPAN_NOTICE("You turn off \the [src]'s ignition."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/cargo_train.dm:205]:
```suggestion
	to_chat(user, SPAN_NOTICE("You turn off \the [src]\s ignition."))
```

File: code/modules/vehicles/cargo_train.dm [hunk 8]
```diff
@@ -243,48 +259,40 @@
...
-	if(Adjacent(user))
-		if(on)
-			stop_engine(usr)
-		else
-			start_engine(usr)
-	else
-		return ..()
+/obj/vehicle/train/cargo/engine/CtrlClick(mob/user)
+	if(load && load != user)
+		to_chat(user, SPAN_WARNING("You can't interact with \the [src] while its in use."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/cargo_train.dm:264]:
```suggestion
		to_chat(user, SPAN_WARNING("You can't interact with \the [src] while it's in use."))
```

File: code/modules/vehicles/cargo_train.dm [hunk 9]
```diff
@@ -243,48 +259,40 @@
...
 
 /obj/vehicle/train/cargo/engine/proc/remove_key(var/mob/user)
 	if(!key)
 		to_chat(usr, SPAN_WARNING("\The [src] doesn't have a key inserted!"))
 		return
 	if(load && load != usr)
+		to_chat(usr, SPAN_WARNING("You can't remove \the [key] from \the [src] while it's in use."))
 		return
-
+	to_chat(user, SPAN_NOTICE("You take out \the [key] out of \the [src]'s ignition."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/cargo_train.dm:294]:
```suggestion
	to_chat(user, SPAN_NOTICE("You take out \the [key] out of \the [src]\s ignition."))
```

File: code/modules/vehicles/train.dm [hunk 10]
```diff
@@ -83,35 +83,18 @@
...
-
-	return 1
 
 /obj/vehicle/train/MouseDrop_T(var/atom/movable/C, mob/user as mob)
-	if(user.buckled_to || user.stat || user.restrained() || !Adjacent(user) || !user.Adjacent(C) || !istype(C) || (user == C && !user.canmove))
+	if(use_check_and_message(user))
 		return
 	if(istype(C, /obj/vehicle/train))
-		attach_to(C, user)
+		latch(C, user) //BANDAID FIX MOMENT. GEEEEVES!!! ALSO GEM YOU DONT NEED TO MAKE A COMMENT TO REMOVE THIS. IT'S NOT LIKE ANYONE'S GOING TO SEE IT ANYWAY. I MEAN. NON PLAYER FACING CHANGES.
```
Comment (SleepyGemmy) [inline code/modules/vehicles/train.dm:91]:
```suggestion
		latch(C, user)
```

File: code/modules/vehicles/train.dm [hunk 11]
```diff
@@ -83,35 +83,18 @@
...
 	else
 		if(!load(C))
-			to_chat(user, "<span class='warning'>You were unable to load [C] on [src].</span>")
+			to_chat(user, SPAN_WARNING("You were unable to load [C] on [src]."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/train.dm:94]:
```suggestion
			to_chat(user, SPAN_WARNING("You were unable to load \the [C] on \the [src]."))
```

File: code/modules/vehicles/train.dm [hunk 12]
```diff
@@ -134,60 +117,68 @@
 //Note: there is a modified version of this in code\modules\vehicles\cargo_train.dm specifically for cargo train engines
 /obj/vehicle/train/proc/attach_to(obj/vehicle/train/T, mob/user)
 	if (get_dist(src, T) > 1)
-		to_chat(user, "<span class='warning'>[src] is too far away from [T] to hitch them together.</span>")
+		to_chat(user, SPAN_WARNING("[src] is too far away from [T] to hitch them together."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/train.dm:120]:
```suggestion
		to_chat(user, SPAN_WARNING("\The [src] is too far away from \the [T] to hitch them together."))
```

File: code/modules/vehicles/train.dm [hunk 13]
```diff
@@ -134,60 +117,68 @@
...
 		return
 
 	if (lead)
-		to_chat(user, "<span class='warning'>[src] is already hitched to something.</span>")
+		to_chat(user, SPAN_WARNING("[src] is already hitched to something."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/train.dm:124]:
```suggestion
		to_chat(user, SPAN_WARNING("\The [src] is already hitched to something."))
```

File: code/modules/vehicles/train.dm [hunk 14]
```diff
@@ -134,60 +117,68 @@
...
 		return
 
 	if (T.tow)
-		to_chat(user, "<span class='warning'>[T] is already towing something.</span>")
+		to_chat(user, SPAN_WARNING("[T] is already towing something."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/train.dm:128]:
```suggestion
		to_chat(user, SPAN_WARNING("\The [T] is already towing something."))
```

File: code/modules/vehicles/train.dm [hunk 15]
```diff
@@ -134,60 +117,68 @@
...
 	//latch with src as the follower
-	lead = T
+	src.lead = T
 	T.tow = src
-	set_dir(lead.dir)
-
-	if(user)
-		to_chat(user, "<span class='notice'>You hitch [src] to [T].</span>")
+	set_dir(get_dir(src, lead))
+	to_chat(user, SPAN_NOTICE("You hitch [src] to [T]."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/train.dm:143]:
```suggestion
	to_chat(user, SPAN_NOTICE("You hitch \the [src] to \the [T]."))
```

File: code/modules/vehicles/train.dm [hunk 16]
```diff
@@ -134,60 +117,68 @@
...
+	playsound(loc, 'sound/items/wrench.ogg', 70, TRUE)
 
 	update_stats()
 
 
 //detaches the train from whatever is towing it
 /obj/vehicle/train/proc/unattach(mob/user)
 	if (!lead)
-		to_chat(user, "<span class='warning'>[src] is not hitched to anything.</span>")
+		to_chat(user, SPAN_WARNING("[src] is not hitched to anything."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/train.dm:152]:
```suggestion
		to_chat(user, SPAN_WARNING("\The [src] is not hitched to anything."))
```

File: code/modules/vehicles/train.dm [hunk 17]
```diff
@@ -134,60 +117,68 @@
...
 		return
 
 	lead.tow = null
 	lead.update_stats()
 
-	to_chat(user, "<span class='notice'>You unhitch [src] from [lead].</span>")
+	to_chat(user, SPAN_NOTICE("You unhitch [src] from [lead]."))
```
Comment (SleepyGemmy) [inline code/modules/vehicles/train.dm:158]:
```suggestion
	to_chat(user, SPAN_NOTICE("You unhitch \the [src] from \the [lead]."))
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**